picoquictest/stream0_frame_test.c

```diff
@@ -132,12 +132,12 @@ static int StreamZeroFrameOneTest(struct test_case_st* test)
     int ret = 0;
 
     picoquic_cnx_t cnx = { 0 };
-    size_t consumed = 0;
     uint64_t current_time = 0;
 
     for (size_t i = 0; ret == 0 && i < test->list_size; i++) {
-        if (0 != picoquic_decode_stream_frame(&cnx, test->list[i].packet,
-                     test->list[i].packet_length, 1, &consumed, current_time)) {
+        const uint8_t* bytes = test->list[i].packet + 1;
+        if (0 != picoquic_decode_stream_frame(&cnx, &bytes, test->list[i].packet + test->list[i].packet_length,
+                                              1, current_time)) {
             FAIL(test, "packet %" PRIst, i);
```

Comment:
I would much rather leave the tests completely unchanged in the first PR. That would minimize the risks of an inadvertent issue. And it will also make the PR smaller.

Reply:
The test is reaching deep into the frame decoder and calling its function that needs an interface change.  It is the test violating the abstraction layer.

Reply:
This is a unit test for the stream 0 frame handling. So yes, it does call specific low level functions -- after all, the purpose is precisely to test them.